contrib/forms/handwritten-notes/hw-notes.php

```diff
@@ -0,0 +1,129 @@
+<?php
+/**
+ * Handwritten Notes Form
+ *
+ * @package   OpenEMR
+ * @link      http://www.open-emr.org
+ * @author    Tyler Wrenn <tyler@tylerwrenn.com>
+ * @copyright Copyright (c) 2020 Tyler Wrenn <tyler@tylerwrenn.com>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ **/
+
+require_once("../../globals.php");
+require_once("$srcdir/classes/Document.class.php");
+
+use OpenEMR\Common\Csrf\CsrfUtils;
+use OpenEMR\Core\Header;
+
+if (!$encounter) {
+    die(xlt("Internal Error: We do not seem to be in an encounter!"));
+}
+
+if ($_POST) {
+    if (!CsrfUtils::verifyCsrfToken($_POST["csrf_token_form"])) {
+        CsrfUtils::csrfNotVerified();
+    }
+}
+    
+if (!empty($_POST['hw-notes'])) {
+    // Create new document to spill the data in
+    $document = new Document();
+        
+    // Replace the unneccessary data in the beginning of the uri
+    $formatted = str_replace("data:application/pdf;filename=generated.pdf;base64,", "", $_POST['hw-notes']);
+        
+    // Set the document to the encounter id
+    $document->set_encounter_id($_SESSION['encounter']);
+
+
+    $doc_id = sqlQuery("SELECT `value` FROM `form_handwritten` WHERE `name` = 'doc_category'");
+
+        
+    $doc = $document->createDocument($pid, $doc_id['value'], 'handwrittennotes-pid' . $pid . '-' . date('m-d-Y-g:ia') . '.pdf', 'application/pdf', base64_decode($formatted), '', '', $_SESSION['authUserID'], null);
+    ?>
+
+<!doctype html>
+<html>
+
+<head>
+    <?php Header::setupHeader(['common']); ?>
+    <title><?php echo xlt("Handwritten Notes"); ?></title>
+    <style>
+        .text-white:active {
+            color: #fff;
+        }
+        .text-white:visited {
+            color: #fff;
+        }
+    </style>
+</head>
+<body class="body_top">
+    <div class="text-center pt-5">
+        <p class="lead"><?php echo xlt("The document has been saved to the patient's document tree."); ?></p>
+        <a href="#" class="btn btn-primary text-white" onclick="window.close()"><?php echo xlt("Close Window"); ?></a>
+    </div>
+    
+</body>
+</html>
+<?php } else { ?>
+<!doctype html>
+<html>
+
+<head>
+    <?php Header::setupHeader(['common', 'jspdf']); ?>
```

Comment:
Does jspdf support utf8 and other languages (like Hebrew and Chinese)?

Reply:
I'm pretty sure it supports UTF8 as this uses JavaScript to generate PDF's. It does support other languages too (in theory) due to the fact that you can create a PDF with anything you want inside of it (apparently it processes Arabic separately [with this method](http://raw.githack.com/MrRio/jsPDF/master/docs/module-arabic.html)). It can also set ISO languages if needed ([referenced here](http://raw.githack.com/MrRio/jsPDF/master/docs/module-setLanguage.html)). The doc also says it has RTL support [shown here](http://raw.githack.com/MrRio/jsPDF/master/docs/jsPDF.html#getR2L). Overall this API is pretty awesome.